osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -137,14 +152,13 @@ public static IEnumerable<(SettingSourceAttribute, PropertyInfo)> GetSettingsSou
             }
         }
 
-        public static IEnumerable<(SettingSourceAttribute, PropertyInfo)> GetOrderedSettingsSourceProperties(this object obj)
+        public static ICollection<(SettingSourceAttribute, PropertyInfo)> GetOrderedSettingsSourceProperties(this object obj)
         {
-            var original = obj.GetSettingsSourceProperties();
-
-            var orderedRelative = original.Where(attr => attr.Item1.OrderPosition != null).OrderBy(attr => attr.Item1.OrderPosition);
-            var unordered = original.Except(orderedRelative);
+            var original = obj.GetSettingsSourceProperties().ToArray();
```

Comment:
I *think* this `ToArray()` shouldn't be required, unless `OrderBy` is doing something really weird?

Reply:
Right, good point. The underlying yield enumerator method looks to have no side effects (at least I'd hope it does), so I'll trim that copy too.